### PR TITLE
fix(core): guard against divide by 0, Infinity, and NaN in `daffDivide`

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: #1844 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/core/src/utils/long-arithmetic.spec.ts [hunk 1]
```diff
@@ -68,6 +68,22 @@ describe('Core | Utils | Long Arithmetic', () => {
...
+      expect(daffDivide(1, null)).toEqual(NaN);
+    });
+
+    it('should return NaN when attempting to divide by 0', () => {
+      expect(daffDivide(0, 0)).toEqual(NaN);
+      expect(daffDivide(1, 0)).toEqual(NaN);
+    });
+
+    it('should return NaN when attempting to perform division including Infinity', () => {
+      expect(daffDivide(1, Infinity)).toEqual(NaN);
```
Comment (damienwebdev) [inline libs/core/src/utils/long-arithmetic.spec.ts:80]: This first case should return `0`.

File: libs/core/src/utils/long-arithmetic.spec.ts [hunk 2]
```diff
@@ -68,6 +68,22 @@ describe('Core | Utils | Long Arithmetic', () => {
...
+      expect(daffDivide(Infinity, 1)).toEqual(NaN);
```
Comment (damienwebdev) [inline libs/core/src/utils/long-arithmetic.spec.ts:81]: This should return `Infinity`

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/core/src/utils/long-arithmetic.spec.ts (hunk 1 above)
Comment (griest024) [reply libs/core/src/utils/long-arithmetic.spec.ts]: changed

File: libs/core/src/utils/long-arithmetic.spec.ts (hunk 2 above)
Comment (griest024) [reply libs/core/src/utils/long-arithmetic.spec.ts]: changed

**Review by damienwebdev — APPROVED** [review]

**Pull request merged**